### PR TITLE
[bsc#1087447] validate pillars

### PR DESCRIPTION
**Pull request by MaximilianMeister**
invalidpillar

### CODE REVIEW

File: app/models/pillar.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # rubocop:disable Metrics/ClassLength
 class Pillar < ApplicationRecord
   validates :pillar, presence: true
-  validates :value, presence: true
+  validates :value, presence: true, format: { with: /\A[\S]+\z/ }
```
Comment (bergmannf) [inline app/models/pillar.rb]: Wouldn't this regular expression force the value to not contain any spaces at all?

That might be problematic for the `username` and `password` of the cloud configurations.

Comment (MaximilianMeister) [reply app/models/pillar.rb]: right, i'll update it

File: app/models/pillar.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # rubocop:disable Metrics/ClassLength
 class Pillar < ApplicationRecord
   validates :pillar, presence: true
-  validates :value, presence: true
+  validates :value, presence: true, format: { with: /\A[\S].*[\S]\z/ }
```
Comment (bergmannf) [inline app/models/pillar.rb]: The second `\[S]` forces every entry to have at least 2 characters, which currently breaks on the pillar `lb_mon_retries: "3"` I think.

**Comment by vitoravelino** [comment]
In the past Rails projects I've worked on we always used https://github.com/rmm5t/strip_attributes to solve this issue. But I'm not sure if it would be fine for Velum since is another gem.

**Comment by jordimassaguerpla** [comment]
I don't see this gem in obs://devel:languages:ruby:extensions. If you want to use that gem, we need to package that one.

I don't see any issue using that, but this will get blocked until we have the rpm.

File: app/models/pillar.rb (hunk 2 above)
Comment (MaximilianMeister) [reply app/models/pillar.rb]: updated, can you test again?

**Comment by MaximilianMeister** [comment]
> I don't see any issue using that, but this will get blocked until we have the rpm.

probably there's no issue, but i think there is also no benefit in functionality. if we use this validator for other fields as well, we might reconsider, but i'd say let's do the minimal now, and save us some gem maintenance

**Pull request merged**